### DISCUSSION TITLE
KOT System Improvements

### DISCUSSION THREAD

**Discussion by rsidhaarth** [Blair2004/NexoPOS#457] [Ideas]
Hi Team,

As of now, you can add any product to the billing screen and then, you can send those to the kitchen. Again the cashier can open that product and cancel that KOT without any reason if it is ready or not ready. Also, there are no records of KOT. So this system should be improved

1) whenever a set of products is being sent to the kitchen, it needs to be assigned with KOT number
2) if the same order receives another product, then it needs to go with an incremental KOT number as other orders may also get multiple KOT so, it needs to be a continuation of that. and so staff can verify what KOT is done and not
3) cancelling the KOT needs a reason and so admin can come to know about it
4) KOT number should be mentioned on the KOT printout and Kitchen Screen
5) All the KOT numbers should be mentioned on the Final Receipt

![image](https://user-images.githubusercontent.com/16856248/147849481-32102bde-84b2-4e1f-84ce-e79e21246f6a.png)

**Comment by Blair2004** [comment]
Products send to the kitchen has a unique ID, which i believe can be considered as a number. I therefore don't believe it's necessary to have a special kot number. 
Regarding cancelation reason, do you mean on Gastro, it's not possible to provide a reason while canceling a meal ?
Kot being mentionned on the receipt, we need an example to take a decision. While i believe this should rather be made as a customization since not all business needs that information on the KOT.

Kind Regards.

**Reply by rsidhaarth** [reply]
This improvement is about two things

1) KOT number to be shown on POS screen with date and time, and so cashier can confirm

- how many KOT has been generated
- can cross-check with the kitchen via KOT numbers
- can confirm which item have ordered at which time and so don't confuse as now only all items are shown together

2) Items can be cancelled only via KOT and so cashier and chef both can be in sync

3) KOT number should be mentioned on the Prinout, KDS and POS screen to confirm all the time
